src/main/jbake/content/blog/2016/10/14/hawkular-apm-polyglot-zipkin.adoc

```diff
@@ -0,0 +1,190 @@
+endif::[]
+
+== Zipkin
+Please start Zipkin server and execute requests from previous section.
+
+The first thing we are going to look at is a query capability. The user interface
+offers to specify service name, names of recorded spans of a given service. 
+It is possible to specify span's start time, end time and duration range. 
+For a more specific query it is possible to query on binary annotations,
+which is helpful if we are looking for a specific spans, for example HTTP URL of invoked service.
+In the following figure it can be seen that we specified URL and HTTP status code 
+of invoked service.
+
+The result of the query shows one trace which consist of 23 spans out of which 39% were 
+reported by `node.js` service (fourth request). There is also a trace duration time and 
+information which services were invoked within that trace (also with a duration if possible 
+to calculate).
+
+
+ifndef::env-github[]
+image::/img/blog/2016/apm-zipkin-createUser-query.png[title="Zipkin Query",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-zipkin-createUser-query.png[title="Zipkin Query",align="center"]
+endif::[]
+
+The next screen is a trace instance view. It shows a timeline with timing data relative 
+to the trace start time. In the following figure there is a trace view for the third call. 
+It can be seen that it represent an asynchronous call. The first call ended before its
+descendants.
+ 
+ifndef::env-github[]
+image::/img/blog/2016/apm-zipkin-async.png[title="Zipkin Trace Detail",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-zipkin-async.png[title="Zipkin Trace Detail",align="center"]
+endif::[]
+
+When you click on a specific span it shows a span detail view with events recorded within
+the span. There are also recorded binary annotations which denotes information like HTTP URL 
+or HTTP status code. Instrumented application can add custom binary annotations.
+
+ifndef::env-github[]
+image::/img/blog/2016/apm-zipkin-async-span-detail.png[title="Zipkin Span Detail",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-zipkin-async-span-detail.png[title="Zipkin Span Detail",align="center"]
+endif::[]
+
+The last Zipkin screen is showing dependencies between services. When you on a specific 
+service it shows how many calls were triggered from a service to a service. The more calls to 
+the service the wider arrow.
+
+ifndef::env-github[]
+image::/img/blog/2016/apm-zipkin-dependencies.png[title="Zipkin Service Dependencies",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-zipkin-dependencies.png[title="Zipkin Service Dependencies",align="center"]
+endif::[]
+
+== Conclusion
+We have seen a brief comparison of Hawkular APM and Zipkin server. Both of these projects 
+displays data differently. Hawkular APM is doing more high level analysis and calculates 
```

Comment:
:+1:  - although separate into two sentences at "statistics. On the other hand ..."